examples/rfcavity/analysis_rfcavity_ref_part.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022-2023 ImpactX contributors
+# Authors: Axel Huebl, Chad Mitchell
+# License: BSD-3-Clause-LBNL
+#
+
+import glob
+import re
+
+import numpy as np
+import pandas as pd
+
+
+def read_file(file_pattern):
+    for filename in glob.glob(file_pattern):
+        df = pd.read_csv(filename, delimiter=r"\s+")
+        if "step" not in df.columns:
+            step = int(re.findall(r"[0-9]+", filename)[0])
+            df["step"] = step
+        yield df
+
+
+def read_time_series(file_pattern):
+    """Read in all CSV files from each MPI rank (and potentially OpenMP
+    thread). Concatenate into one Pandas dataframe.
+
+    Returns
+    -------
+    pandas.DataFrame
+    """
+    return pd.concat(
+        read_file(file_pattern),
+        axis=0,
+        ignore_index=True,
+    )  # .set_index('id')
+
+
+# read reference particle data
+rbc = read_time_series("diags/ref_particle.*")
+
+s = rbc["s"]
+gamma = rbc["gamma"]
+
+si = s.iloc[0]
+gammai = gamma.iloc[0]
+
+length = len(s) - 1
+
+sf = s.iloc[length]
+gammaf = gamma.iloc[length]
```

Comment:
I think `-1` universally refers to the last element in Python arrays:
```suggestion
sf = s.iloc[-1]
gammaf = gamma.iloc[-1]
```